### PR TITLE
Generate SubmissionEntry from KV export

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This is the first part of #13169
    
In this PR we:
    
- (in lf land) introduce a simple protobuf to store submission entry (which roughly correspond to TRANSACTION_ENTRY and PACKAGE_UPLOAD_ENTRY KV submission)
  
- (in kv land) we create a tool to extract form a ledger export a sequence of submission entry.
  This will be moved out of the main repo into the oem-integratio-kit
    
CHANGELOG_BEGIN
CHANGELOG_END

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by remyhaemmerle-da** [comment]
Any suggestion to improve terminology is welcomed.

File: ledger/participant-state/kvutils/tools/BUILD.bazel [hunk 1]
```diff
@@ -170,6 +170,20 @@ da_scala_library(
     ],
 )
 
+da_scala_binary(
+    name = "entries-extractor",
```
Comment (hubert-da) [inline ledger/participant-state/kvutils/tools/BUILD.bazel:174]: `transaction-entry-extractor`?

File: ledger/participant-state/kvutils/tools/entries-extractor/src/replay/scala/ledger/participant/state/kvutils/tools/extractor/Extractor.scala [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+          .to(LazyList)
+          .map(_.getSubmission)
+          .flatMap(submissionEnvelope =>
+            decodeEnvelope(participantId, Raw.Envelope(submissionEnvelope))
+          )
+      case Envelope.LogEntryMessage(_) | Envelope.StateValueMessage(_) =>
+        LazyList.empty
+    }
+
+  private[this] def decodeSubmissionInfo(submissionInfo: SubmissionInfo) =
```
Comment (hubert-da) [inline ledger/participant-state/kvutils/tools/entries-extractor/src/replay/scala/ledger/participant/state/kvutils/tools/extractor/Extractor.scala:78]: I think the methods should be in the opposite order to follow the execution order. This method, for instance, is the first one, as it's being run by the app/main.

File: ledger/participant-state/kvutils/tools/entries-extractor/src/replay/scala/ledger/participant/state/kvutils/tools/extractor/Extractor.scala [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import com.daml.ledger.participant.state.kvutils.{Envelope, Raw}
+import com.daml.lf.data._
+import com.daml.lf.testing.snapshot.Snapshot
+
+import java.io.BufferedOutputStream
+import java.nio.file.{Files, Path, Paths}
+import scala.jdk.CollectionConverters._
+import scala.util.control.NonFatal
+
+object Extractor extends App {
```
Comment (hubert-da) [inline ledger/participant-state/kvutils/tools/entries-extractor/src/replay/scala/ledger/participant/state/kvutils/tools/extractor/Extractor.scala:21]: Do you plan to add some tests for it?

File: ledger/participant-state/kvutils/tools/BUILD.bazel (hunk 1 above)
Comment (remyhaemmerle-da) [reply ledger/participant-state/kvutils/tools/BUILD.bazel:174]: :+1:

File: ledger/participant-state/kvutils/tools/entries-extractor/src/replay/scala/ledger/participant/state/kvutils/tools/extractor/Extractor.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply ledger/participant-state/kvutils/tools/entries-extractor/src/replay/scala/ledger/participant/state/kvutils/tools/extractor/Extractor.scala:78]: :+1:

File: ledger/participant-state/kvutils/tools/entries-extractor/src/replay/scala/ledger/participant/state/kvutils/tools/extractor/Extractor.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply ledger/participant-state/kvutils/tools/entries-extractor/src/replay/scala/ledger/participant/state/kvutils/tools/extractor/Extractor.scala:21]: Unfortunate I cannot test it easily in the main repo, as there is not more ledger export generated here.
Once you move the tool to the integration kit, I will be please to write tests there. 

Note I have tested manually.

**Review by cocreature — APPROVED** [review]
Looks sensible,  thanks! It might be good to have a corresponding library that reads it again (which would live in the daml repo) and have some roundtrip tests (in the kvutils repo) that check that when you can decode the things produced by this again.

**Pull request merged**